src/libs/actions/Report.js

```diff
@@ -1237,19 +1234,22 @@ function subscribeToNewActionEvent(reportID, callback) {
  * @param {Object} action
  */
 function viewNewReportAction(reportID, action) {
+    const report = allReports[reportID];
     const isFromCurrentUser = action.actorAccountID === currentUserAccountID;
     const updatedReportObject = {};
 
-    // When handling an action from the current user we can assume that their last read actionID has been updated in the server,
-    // but not necessarily reflected locally so we will update the lastReadSequenceNumber to mark the report as read.
-    updatedReportObject.maxSequenceNumber = action.sequenceNumber;
+    // When handling an action from the current user we can assume that their lastReadMessage has been updated in the server,
+    // but not necessarily reflected locally so we will update the lastMessageTimestamp to mark the report as read.
+    if (report.lastMessageTimestamp < action.reportActionTimestamp) {
+        updatedReportObject.lastMessageTimestamp = action.reportActionTimestamp;
```

Comment:
The comment makes me think that this should actually be looking at the `lastReadTimestamp` instead of the `lastMessageTimestamp`.